src/database/mod.rs

```diff
@@ -133,8 +155,11 @@ impl<'r> sqlx::Decode<'r, sqlx::Postgres> for ApprovalStatus {
             <(Option<String>, Option<String>) as sqlx::Decode<sqlx::Postgres>>::decode(value)?;
 
         match (approver, sha) {
-            (Some(approver), Some(sha)) => Ok(ApprovalStatus { approver, sha }),
-            _ => Err("Cannot deserialize ApprovalStatus, both approve and SHA is required".into()),
+            (Some(approver), Some(sha)) => {
+                Ok(ApprovalStatus::Approved(ApprovalInfo { approver, sha }))
+            }
+            (None, None) => Ok(ApprovalStatus::NotApproved),
+            _ => Err("Incosistent DB approval status".into()),
```

Comment:
`Incosistent` -> `Inconsistent` (yeah I know it's my typo xD I meant it as a temporary string, but it's enough I guess)